File: blind75/12_number_of_1_bits/number_of_1_bits.py
def hamming_weight(n: int) -> int:
    """
    Returns the number of '1' bits in the binary representation.
    Uses Brian Kernighan's algorithm for optimal performance.

    Args:
        n: A positive integer

    Returns:
        The count of set bits (Hamming weight)
    """
    count = 0

    while n != 0:
        n = n & (n - 1)  # Clear the rightmost set bit
        count += 1

    return count

File: blind75/12_number_of_1_bits/number_of_1_bits.go
package number_of_1_bits

// HammingWeight returns the number of '1' bits in the binary representation.
// Uses Brian Kernighan's algorithm for optimal performance.
func HammingWeight(n int) int {
	count := 0

	for n != 0 {
		n = n & (n - 1) // Clear the rightmost set bit
		count++
	}

	return count
}
